### PR TITLE
goasm: Full amd64 implementation

### PR DESCRIPTION
**Pull request by pjbgf**
Implements amd64 version which can loop through `p` blocks, instead of being called for each block.

The improvements over partial goasm implementation does not justify the code complexity required for the full implementation:
```
BenchmarkHash8Bytes           4MB/s 
BenchmarkHash8K              12MB/s
BenchmarkHashWithCollision  150MB/s
```

The collision blocks is where the majority of the performance improvements can be seen, however those are meant to be the edge case.

:warning: The `checkubc` block is not finished - hence why the tests for collision are not passing.

### CODE REVIEW

File: sha1cdblock_amd64.s [hunk 1]
```diff
@@ -1,2274 +1,2470 @@
...
+
+checkubc:
+	// There is a total of 33 DvInfo instances in the sha1_dvs array (344 bytes each).
+	MOVL $0, dvi-1992(SP)
+
+	// TODO: the JMP below is to avoid running the checkubc code that is currently
+	// not working.
+	JMP nextblock
+
+dv_loop:
```
Comment (pjbgf) [inline sha1cdblock_amd64.s:2238]: `checkubc` and `dv_loop` requires a bit more TLC in order to work.
